tools/box/box.cc

```diff
@@ -315,13 +315,16 @@ jxl::Status DecodeJpegXlToJpeg(jxl::DecompressParams params,
     if (jpeg_data->app_marker_type[i] == jxl::jpeg::AppMarkerType::kXMP) {
       if (have_xmp)
         return JXL_FAILURE("Unexpected: more than one XMP box required?");
-      if (jpeg_data->app_data[i].size() != container.xml[0].second + 32) {
-        return JXL_FAILURE(
-            "XMP box size does not match JPEG reconstruction data");
+      // Container xml might be empty if xmp data not in xml box.
+      if (!container.xml.empty()) {
+        if (jpeg_data->app_data[i].size() != container.xml[0].second + 32) {
+          return JXL_FAILURE(
+              "XMP box size does not match JPEG reconstruction data");
```

Comment:
This error (or a variant saying there is no XMP box at all) should also be returned in case XMP is empty while jpeg reconstruction expects to have one.

Not that it matters too much, this is legacy code that will become redundant when djxl gets replaced by djxl_ng.